main_loop.py

```diff
@@ -0,0 +1,150 @@
+"""
+MODES:
+[31] No code on robot:
+RoboRIO inactive/not receiving serial data from RoboRIO
+
+[32] Disabled without auto
+RoboRIO is disabled, and has NO autonomous program set
+
+[33] Disabled with auto
+RoboRIO is disabled, and DOES have autonomous program set
+
+
+--< Modes for when RoboRIO is enabled >--
+Priority: 1 > 2 > 3 > 4 > 34
+
+[34] Enabled
+RoboRIO is enabled, but none of the above are active.
+
+[1] Signal:
+For communication with teammates to activate amplifier
+
+[2] Ready to shoot:
+The robot shooter and arm is in position, ready to shoot note
+
+[3] Has note:
+The robot currently is holding a note
+
+[4] Vision:
+The robot spots a note (tells the driver that they can press a button for
+robot to retrieve a note)
+"""
+
+
+import supervisor
+import board
+from lightstrip import LightStrip
+import patterns
+import time
+import sys
+
+# Serial data variables
+serial = sys.stdin
+tolerance = 2
+tolerance_count = 0
+current_mode = "31"
```

Comment:
⭐ ⭐ instead of hardcoding these to strings, if you created named variables for these it would be so much nicer to read this as:
```python
current_mode = NO_CODE
```

Reply:
or seeing the code you have below it could be
```python
current_mode = modes["NO_CODE"]